lib/streamlit/git_util.py

```diff
@@ -106,3 +119,24 @@ def get_repo_info(self):
             return None
 
         return repo, branch, self.module
+
+    def get_uncommitted_files(self):
+        if not self.is_valid():
+            return None
+
+        return [item.a_path for item in self.repo.index.diff(None)]
+
+    def get_ahead_commits(self):
+        if not self.is_valid():
+            return None
+
+        try:
+            remote, branch_name = self.get_tracking_branch_remote()
+
+            return list(
+                self.repo.iter_commits(
+                    "/".join([remote.name, branch_name]) + ".." + branch_name
```

Comment:
Another small nit, but I'd probably just create a variable like `remote_branch = "/".join(...)`

You can also use python [formatted strings](https://realpython.com/python-formatted-output/#the-python-formatted-string-literal-f-string) It makes this really clean.

`f'{remote_branch}..{branch_name}'`